src/swarm/neo/client/mixins/DeserializeMethods.d

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+
+    Mixin for methods to deserialize a received record.
+
+    Copyright:
+        Copyright (c) 2017 sociomantic labs GmbH. All rights reserved
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module swarm.neo.client.mixins.DeserializeMethods;
+
+import ocean.util.serialize.contiguous.MultiVersionDecorator;
+
+/*******************************************************************************
+
+    Mixin for methods to deserialize a received record.
+
+    Params:
+        src = the buffer to be deserialized
+
+*******************************************************************************/
+
+template DeserializeMethods ( alias src )
+{
+    import ocean.util.serialize.contiguous.Contiguous;
+    import ocean.util.serialize.Version;
+    import ocean.util.serialize.contiguous.Deserializer;
+    import ocean.util.serialize.contiguous.MultiVersionDecorator;
+
+    /***************************************************************************
+
+        Deserializes `src` into `dst`, using the provided version decorater, if
+        required.
+
+        Params:
+            T = type of struct to deserialize
+            dst = deserialization destination
+
+        Returns:
+            deserialized struct
+
+    ***************************************************************************/
+
+    public Contiguous!(T) deserialize ( T ) ( ref Contiguous!(T) dst )
+    {
+        static if ( Version.Info!(T).exists )
+        {
+            if ( version_decorator is null )
+                version_decorator = new VersionDecorator;
```

Comment:
Is it for anybody else confusing how `VersionDecorator.VersionDecorator` (deprecated) and `MultiVersionDecorator.VersionDecorator` share the same name, but they are imported from the different modules?